src/rust/engine/process_execution/docker/src/docker.rs

```diff
@@ -184,10 +192,47 @@ pub enum ImagePullPolicy {
   OnlyIfLatestOrMissing,
 }
 
+async fn credentials_for_image(
+  executor: &Executor,
+  image: &str,
+) -> Result<Option<DockerCredentials>, String> {
+  let Some((server, _)) = image.split_once("/") else {
```

Comment:
This seems pretty mysterious. Can you link a spec that proves this makes sense in a comment?

Reply:
The grammar for image references is apparently here: https://github.com/distribution/distribution/blob/main/reference/reference.go

Reply:
The [`docker_image_reference` crate](https://docs.rs/docker-image-reference/0.1.0/docker_image_reference/) might be relevant.

Reply:
I was using the definition of `Repository` from https://github.com/moby/moby/blob/master/image/spec/v1.2.md#terminology ... but staring at these again, I think that I need to harden this a bit to confirm that the prefix is actually a DNS name.

Reply:
Whatever the outcome it would be great to leave a link to the relevant spec.